File: src/main/old/bbarena/model/pitch/TeamSetUp.java
package bbarena.model.pitch;

import bbarena.model.Choice;
import bbarena.model.Coordinate;

import java.util.HashMap;
import java.util.Map;

public class TeamSetUp implements Choice {

    private final Map<Long, Coordinate> _playersPlacement;

    public TeamSetUp() {
        _playersPlacement = new HashMap<>();
    }

    public void placePlayer(long playerId, Integer x, Integer y) {
        _playersPlacement.put(playerId, new Coordinate(x, y));
    }

    public Map<Long, Coordinate> getSetUp() {
        return _playersPlacement;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        return true;
    }

    @Override
    public int hashCode() {
        return getClass().hashCode();
    }
}

File: src/main/kotlin/bbarena/model/pitch/TeamSetUp.kt
package bbarena.model.pitch

import bbarena.model.Choice
import bbarena.model.Coordinate
import java.util.*

class TeamSetUp : Choice {

    private val _playersPlacement: MutableMap<Long, Coordinate>

    val setUp: Map<Long, Coordinate>
        get() = _playersPlacement

    init {
        _playersPlacement = HashMap()
    }

    fun placePlayer(playerId: Long, x: Int?, y: Int?) {
        _playersPlacement[playerId] = Coordinate(x!!, y!!)
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        return if (o == null || javaClass != o.javaClass) false else true

    }

    override fun hashCode(): Int {
        return javaClass.hashCode()
    }
}
